packages/cli/src/endpoints/credentials.endpoints.ts

```diff
@@ -0,0 +1,385 @@
+
+			// Add special database related data
+			newCredentialData.updatedAt = new Date();
+
+			await this.externalHooks.run('credentials.update', [newCredentialData]);
+
+			// Update the credentials in DB
+			await Db.collections.Credentials!.update(credentialId, newCredentialData);
+
+			// We sadly get nothing back from "update". Neither if it updated a record
+			// nor the new value. So query now the hopefully updated entry.
+			const responseData = await Db.collections.Credentials!.findOne(credentialId);
+
+			if (responseData === undefined) {
+				throw new ResponseHelper.ResponseError(
+					`Credential ID "${credentialId}" could not be found to be updated.`,
+					undefined,
+					400,
+				);
+			}
+
+			// Remove the encrypted data as it is not needed in the frontend
+			const { id, data, ...rest } = responseData;
+
+			return {
+				id: id.toString(),
+				...rest,
+			};
+		}),
+	);
+
+	/**
+	 * Retrieve a credential.
+	 */
+	this.app.get(
+		`/${this.restEndpoint}/credentials/:id`,
+		ResponseHelper.send(async (req: CredentialRequest.Get) => {
+			const { id: credentialId } = req.params;
+
+			const shared = await Db.collections.SharedCredentials!.findOne({
+				relations: ['credentials'],
+				where: whereClause({
+					user: req.user,
+					entityType: 'credentials',
+					entityId: credentialId,
+				}),
+			});
+
+			if (!shared) return {};
+
+			const { credentials: credential } = shared;
+
+			if (req.query.includeData !== 'true') {
+				const { data, id, ...rest } = credential;
+
+				return {
+					id: id.toString(),
+					...rest,
+				};
+			}
+
+			const { data, id, ...rest } = credential;
+
+			const encryptionKey = await UserSettings.getEncryptionKey();
+
+			if (!encryptionKey) {
+				throw new ResponseHelper.ResponseError(
+					RESPONSE_ERROR_MESSAGES.NO_ENCRYPTION_KEY,
+					undefined,
+					500,
+				);
+			}
+
+			const coreCredential = new Credentials(
+				{ id: credential.id.toString(), name: credential.name },
+				credential.type,
+				credential.nodesAccess,
+				credential.data,
+			);
+
+			return {
+				id: id.toString(),
+				data: coreCredential.getData(encryptionKey),
+				...rest,
+			};
+		}),
+	);
+
+	/**
+	 * Retrieve all credentials.
+	 */
+	this.app.get(
+		`/${this.restEndpoint}/credentials`,
+		ResponseHelper.send(async (req: CredentialRequest.GetAll): Promise<ICredentialsResponse[]> => {
+			let credentials: ICredentialsDb[] = [];
+
+			const filter = req.query.filter
+				? (JSON.parse(req.query.filter) as Record<string, string>)
+				: {};
+
+			if (req.user.globalRole.name === 'owner') {
+				credentials = await Db.collections.Credentials!.find({
+					select: ['id', 'name', 'type', 'nodesAccess', 'createdAt', 'updatedAt'],
+					where: filter,
+				});
+			} else {
+				const shared = await Db.collections.SharedCredentials!.find({
+					relations: ['credentials'],
+					where: whereClause({
+						user: req.user,
+						entityType: 'credentials',
+					}),
+				});
```

Comment:
I think for the shared-entities we need to make the individual keys available, because here we are joining the whole credentials in order to just get the id and fetch them again. Same for workflows at multiple points

(can be done later)

Reply:
Implemented with `@RelationId`. Let me know if that's what you meant.